api/tests/test_room_archive.py

```diff
@@ -0,0 +1,85 @@
+from asyncio import CancelledError
+from typing import AsyncIterator, Callable, Iterator
+from uuid import uuid4
+
+import aiobotocore
+import pytest
+from aiobotocore.client import AioBaseClient
+
+from localstack.services import infra
+from pytest_lazyfixture import lazy_fixture
+
+from src.room_store.room_archive import RoomArchive
+from src.room_store.s3_room_archive import S3RoomArchive
+from src.util.async_util import async_collect
+from tests.static_fixtures import VALID_ACTION
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture(scope='session')
+def s3_url() -> Iterator[str]:
+    infra.start_infra(asynchronous=True, apis=['s3'])
+    yield 'http://localhost:4566'
+    infra.stop_infra()
+
+
+@pytest.fixture
+async def s3_client(s3_url: str) -> AsyncIterator[AioBaseClient]:
+    s3_client_context = aiobotocore.get_session().create_client(
+        's3',
+        region_name='us-east-1',
+        endpoint_url=s3_url,
+        aws_access_key_id='test',
+        aws_secret_access_key='test',
+    )
+    try:
+        async with s3_client_context as s3_client:
+            yield s3_client
+    except CancelledError:
+        pass
```

Comment:
Good candidate for https://docs.python.org/3/library/contextlib.html#contextlib.suppress